pkg/storage/unified/index/indexer.go

```diff
@@ -0,0 +1,66 @@
+package index
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/blevesearch/bleve/v2"
+	"github.com/blevesearch/bleve/v2/mapping"
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+	"google.golang.org/grpc"
+)
+
+type IndexConfig struct {
+	ResourceClient resource.ResourceClient
+}
+
+// IndexFromBeginning indexes selected fields and indexes them, for all resources
+// when the server starts.
+func (i IndexConfig) IndexFromBeginning(ctx context.Context) error {
+	m := createIndexMapping()
+	index, err := bleve.NewMemOnly(m)
+	if err != nil {
+		return fmt.Errorf("could not create new bleve index: %w", err)
+	}
+	// for now listing and indexing syncronously, we can improve this later
+	var allResources *resource.ListResponse
+	l, err := i.listAllResources(ctx, allResources)
```

Comment:
I didn't realize there are so few fields that are useful on the ListResponse 😢

Is there a generic resource object we can unmarshal the items into first?